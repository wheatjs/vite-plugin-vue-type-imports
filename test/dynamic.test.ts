import { readFile } from "node:fs/promises";
import { resolve } from "node:path";
import { CodeGetter, defineTransformTest } from "./_utils";

const codeGetter: CodeGetter = async ({ entry }) => readFile(resolve(__dirname, entry), 'utf-8');

const structureRE = /.+\/dynamic\/(.+)\/(.+)\//g

await defineTransformTest({
    category: 'Dynamic',
    filePattern: ['./fixtures/dynamic/**/*.vue'],
    fileName: __filename,
    codeGetter,
    structureRE
})
