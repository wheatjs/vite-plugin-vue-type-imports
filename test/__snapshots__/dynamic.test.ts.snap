// Vitest Snapshot v1

exports[`Dynamic > Interface extends interface > No reference > index.vue (clean all) 1`] = `
"<script lang=\\"ts\\" setup>
interface Props {
  baz: boolean

  foo: string
  bar: number
}
defineProps<Props>()
</script>
"
`;

exports[`Dynamic > Interface extends interface > No reference > index.vue (clean interface) 1`] = `
"<script lang=\\"ts\\" setup>
interface Props {
  baz: boolean

  foo: string
  bar: number
}





defineProps<Props>()

</script>
"
`;

exports[`Dynamic > Interface extends interface > No reference > index.vue (clean newline) 1`] = `
"<script lang=\\"ts\\" setup>
interface Props {
  baz: boolean

  foo: string
  bar: number
}
defineProps<Props>()
</script>
"
`;

exports[`Dynamic > Interface extends interface > No reference > index.vue (default) 1`] = `
"<script lang=\\"ts\\" setup>
interface Props {
  baz: boolean

  foo: string
  bar: number
}





defineProps<Props>()

</script>
"
`;

exports[`Dynamic > Interface extends interface > No reference > internal.vue (clean all) 1`] = `
"<script lang=\\"ts\\" setup>
interface Props {
  baz: boolean

  foo: string
  bar: number
}
defineProps<Props>()
</script>
"
`;

exports[`Dynamic > Interface extends interface > No reference > internal.vue (clean interface) 1`] = `
"<script lang=\\"ts\\" setup>
interface Props {
  baz: boolean

  foo: string
  bar: number
}





defineProps<Props>()

</script>
"
`;

exports[`Dynamic > Interface extends interface > No reference > internal.vue (clean newline) 1`] = `
"<script lang=\\"ts\\" setup>
interface Props {
  baz: boolean

  foo: string
  bar: number
}
interface BaseProps {
  baz: boolean
}
defineProps<Props>()
</script>
"
`;

exports[`Dynamic > Interface extends interface > No reference > internal.vue (default) 1`] = `
"<script lang=\\"ts\\" setup>
interface Props {
  baz: boolean

  foo: string
  bar: number
}

interface BaseProps {
  baz: boolean
}



defineProps<Props>()

</script>
"
`;

exports[`Dynamic > Interface without reference > No transform > index.vue (clean all) 1`] = `
"<script lang=\\"ts\\" setup>
interface Props {
  foo: string
  bar: number
}

defineProps<Props>()
</script>
"
`;

exports[`Dynamic > Interface without reference > No transform > index.vue (clean interface) 1`] = `
"<script lang=\\"ts\\" setup>
interface Props {
  foo: string
  bar: number
}

defineProps<Props>()
</script>
"
`;

exports[`Dynamic > Interface without reference > No transform > index.vue (clean newline) 1`] = `
"<script lang=\\"ts\\" setup>
interface Props {
  foo: string
  bar: number
}

defineProps<Props>()
</script>
"
`;

exports[`Dynamic > Interface without reference > No transform > index.vue (default) 1`] = `
"<script lang=\\"ts\\" setup>
interface Props {
  foo: string
  bar: number
}

defineProps<Props>()
</script>
"
`;
